lib/storage.js

```diff
@@ -13,18 +13,26 @@
 'use strict';
 
 const Source = require('./source');
-const { AssetComputeLogUtils } = require('@adobe/asset-compute-commons');
+const Watermark = require('./watermark');
+const { AssetComputeLogUtils, RenditionFormatUnsupportedError } = require('@adobe/asset-compute-commons');
 const http = require('./storage/http');
 const datauri = require('./storage/datauri');
 const URL = require('url');
 const fs = require("fs-extra");
 const path = require('path');
+const fileType = require('file-type');
 
 const DATA_PROTOCOL = 'data:';
 
-function isValidLocalFile(fileName, basePath){
+/* TODO: refactor to check file type as a stream before downloading?
```

Comment:
No need to check as stream.

Reply:
Start with the easier check to verify it's a PNG file after download.

Reply:
Needs a ticket for later?